components/server/src/server.rs

```diff
@@ -1257,3 +1263,33 @@ impl<R: RaftEngine> EngineMetricsManager<R> {
         }
     }
 }
+
+#[derive(Default)]
+pub struct MemoryTraceManager {
+    providers: Vec<Arc<MemoryTraceNode>>,
+}
+
+impl MemoryTraceManager {
+    pub fn flush(&mut self, _duration: Duration) {
+        for provider in &self.providers {
+            let provider_name = provider.readable_name();
+            let ids = provider.get_children_ids();
+            for id in ids {
+                let sub_trace = provider.sub_trace(id);
+                let sub_trace_name = sub_trace.readable_name();
```

Comment:
`readable_name` may not good to be a metric name since words are separated by space.

Reply:
Better to add a function like `original_name`?